### PR TITLE
First Pass of WebLab Tests

### PR DESCRIPTION
**Pull request by JillianK**
I added tests for WebLab init, getCodeAsync, onProjectChanged, and setBrambleHost.

Some things that probably should be tested but I couldn't figure out how to get to (please let me know if there's any I'm missing in this list):
- config.afterClearPuzzle in init
- onMount in init
- onStartFullScreenPreview (also in init)
- onToggleInspector (also in init)
- onFilesReady in loadFileEntries

I'm not sure if we need tests for these but I also couldn't figure out how to get to:
- deleteProjectFile
- renameProjectFile
- changeProjectFile

## Links

- jira ticket: [STAR-1448](https://codedotorg.atlassian.net/browse/STAR-1448?atlOrigin=eyJpIjoiNTgwNTFjZWVjYjhhNDJkZDlmNzFjZWZmYzIwNTQ2NGEiLCJwIjoiaiJ9)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test/unit/weblab/WebLabTest.js [hunk 1]
```diff
@@ -124,4 +201,80 @@ describe('WebLab', () => {
...
   });
+
+  describe('getCodeAsync', () => {
+    it('resolves with empty string if brambleHost is null', () => {
+      weblab.brambleHost = null;
+      weblab.getCodeAsync().then(value => {
+        expect(value).to.equal('');
+      });
+    });
+    it('rejects with error if brambleHost syncFiles has an error', () => {
```
Comment (maddiedierker) [inline apps/test/unit/weblab/WebLabTest.js:205]: nit: add an empty newline between tests for readability

File: apps/test/unit/weblab/WebLabTest.js [hunk 2]
```diff
@@ -12,6 +28,67 @@ describe('WebLab', () => {
...
+      weblab.studioApp_ = null;
+      expect(weblab.init).to.throw(Error);
+    });
+
+    it('dispatches changeMaxProjectCapacity', () => {
+      weblab.init(config);
+      expect(getStore().dispatch).to.have.been.calledWith(
+        changeMaxProjectCapacity(20971520)
+      );
+      expect(ReactDOM.render).to.have.been.calledOnce;
```
Comment (maddiedierker) [inline apps/test/unit/weblab/WebLabTest.js]: i think this was leftover from us pairing 😄 this expectation is unrelated to the test, so we should remove it

File: apps/test/unit/weblab/WebLabTest.js [hunk 3]
```diff
@@ -12,6 +28,67 @@ describe('WebLab', () => {
...
+      config.level.startSources = '{:';
+      expect(() => weblab.init(config)).to.throw(Error);
+      expect(weblab.startSources).to.be.undefined;
+    });
+
+    it('sets startSources if given valid JSON', () => {
+      const validJSON = {value: 'test'};
+      config.level.startSources = JSON.stringify(validJSON);
+      weblab.init(config);
+      expect(weblab.startSources.value).to.equal('test');
```
Comment (maddiedierker) [inline apps/test/unit/weblab/WebLabTest.js]: this expectation might be clearer if we do a "deep equals" comparison:
```javascript
expect(weblab.startSources).to.deep.equal({value: 'test'});
```

File: apps/test/unit/weblab/WebLabTest.js [hunk 4]
```diff
@@ -12,6 +28,67 @@ describe('WebLab', () => {
...
+    });
+
+    it('calls the filesApi to get files', () => {
+      sinon.spy(filesApi, 'getFiles');
+      weblab.init(config);
+      expect(filesApi.getFiles).to.have.been.calledOnce;
+    });
```
Comment (maddiedierker) [inline apps/test/unit/weblab/WebLabTest.js]: this test is actually testing `loadFileEntries`, which has enough logic that it should probably have its own set of tests (e.g., it's own `describe` block with multiple test cases)

File: apps/test/unit/weblab/WebLabTest.js [hunk 5]
```diff
@@ -124,4 +201,80 @@ describe('WebLab', () => {
...
+      weblab.brambleHost = null;
+      weblab.getCodeAsync().then(value => {
+        expect(value).to.equal('');
+      });
+    });
+    it('rejects with error if brambleHost syncFiles has an error', () => {
+      weblab.brambleHost = {
+        syncFiles: callback => {
+          callback('error');
+        }
```
Comment (maddiedierker) [inline apps/test/unit/weblab/WebLabTest.js]: nit: this expression can be simplified to `syncFiles: callback => callback('error')`

Comment (maddiedierker) [reply apps/test/unit/weblab/WebLabTest.js]: (same on lines 224-226 below)

File: apps/test/unit/weblab/WebLabTest.js (hunk 4 above)
Comment (JillianK) [reply apps/test/unit/weblab/WebLabTest.js]: Ok, I'll take that out for now and make loadFileEntries its own set of tests in the next pass

**Review by maddiedierker — APPROVED** [review]
🎉 🎉 🎉  it's so great to see this get test coverage!
